cypher-shell/src/main/java/org/neo4j/shell/CypherShell.java

```diff
@@ -159,30 +155,32 @@ void executeCmd(@Nonnull final CommandExecutable cmdExe) throws ExitException, C
 
     /**
      * Open a session to Neo4j
+     *
+     * @param connectionConfig
      */
     @Override
-    public void connect(@Nonnull final String host, final int port, @Nonnull final String username,
-                        @Nonnull final String password) throws CommandException {
+    public void connect(ConnectionConfig connectionConfig) throws CommandException {
         if (isConnected()) {
             throw new CommandException(String.format("Already connected. Call @|bold %s|@ first.",
                     Disconnect.COMMAND_NAME));
         }
 
         final AuthToken authToken;
-        if (username.isEmpty() && password.isEmpty()) {
+        if (connectionConfig.username().isEmpty() && connectionConfig.password().isEmpty()) {
             authToken = null;
-        } else if (!username.isEmpty() && !password.isEmpty()) {
-            authToken = AuthTokens.basic(username, password);
-        } else if (username.isEmpty()) {
+        } else if (!connectionConfig.username().isEmpty() && !connectionConfig.password().isEmpty()) {
+            authToken = AuthTokens.basic(connectionConfig.username(), connectionConfig.password());
+        } else if (connectionConfig.username().isEmpty()) {
+            //TODO PRAVEENA : When does this get called?
```

Comment:
It is called like the exception messages say, if only a password is specified. Note that all if-statements above check that  *both* username and password is set, or not set. any other case (below) are user errors.